src/main/java/com/odysseusinc/arachne/datanode/controller/BaseDataSourceController.java

```diff
@@ -189,103 +240,41 @@ protected CommonModelType checkDataSource(DataSource dataSource) {
             return setValidationErrors(bindingResult);
         }
         final User user = getAdmin(principal);
-        final DataSource dataSource = convert(dataSourceDTO);
+        DataSource dataSource = conversionService.convert(dataSourceDTO, DataSource.class);
         dataSource.setId(id);
-        final DataSource savedDataSource = dataSourceService.update(user, dataSource);
-        return new JsonResult<>(NO_ERROR, conversionService.convert(savedDataSource, DataSourceDTO.class));
-    }
-
-    @ApiOperation(value = "Register datasource on arachne central")
-    @RequestMapping(
-            value = Constants.Api.DataSource.CENTRAL_REGISTER,
-            method = RequestMethod.POST,
-            produces = MediaType.APPLICATION_JSON_VALUE
-    )
-    public JsonResult<BusinessDTO> register(
-            Principal principal,
-            @PathVariable("id") Long id,
-            @Valid @RequestBody BusinessDTO dataSourceBusinessDTO
-    ) throws PermissionDeniedException {
-
-        return processDataSource(principal, id, dataSourceBusinessDTO, (parameters) -> {
-            final JsonResult<CommonDTO> res = integrationService.sendDataSourceRegistrationRequest(
-                    parameters.user,
-                    parameters.dataSource.getDataNode(),
-                    parameters.commonDTO
-            );
-            if (NO_ERROR.getCode().equals(res.getErrorCode())) {
-                dataSourceService.markDataSourceAsRegistered(parameters.dataSource, res.getResult().getId());
-            }
-            return res;
-        });
-    }
 
+        String inputPassword = dataSourceDTO.getDbPassword();
+        dataSource.setPassword(isNotDummyPassword(inputPassword)? inputPassword: dataSourceService.getById(id).getPassword());
 
-    @ApiOperation(value = "Unregister datasource on arachne central")
-    @RequestMapping(
-            value = Constants.Api.DataSource.CENTRAL_REGISTER,
-            method = RequestMethod.DELETE,
-            produces = MediaType.APPLICATION_JSON_VALUE
-    )
-    public JsonResult unregister(@PathVariable("id") Long id) {
+        AutoDetectedFields autoDetectedFields = checkDataSource(dataSource);
+        final DataSource savedDataSource = dataSourceService.update(user, dataSource);
```

Comment:
Why cannot you first update DS, get back DS with not hidden pass and then do auto-detection?